### DISCUSSION TITLE
C# Optimized Keyboard

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#6884] [Show and tell]
Four months ago I bought a "programmer dvorak" keyboard from WASD Keyboards.  I loved it, but I write a lot of code and play piano and I couldn't help feeling like some of the layout could have been done better.

So I spent the last four months experimenting with different key positions and I'm excited to reveal V1 of my ```Dvorak#``` keyboard.
It has been specifically optimized for C#.

![image](https://user-images.githubusercontent.com/13609558/212522694-168f4220-618e-4d8f-8d0c-f32d9c79f130.png)

Here are a few things I'm pretty happy about:
**"Grouping" constructs follow a common pattern.  Open, Something, Close** 
- ( * )
- [ ! ]
- < = >
- { : }
- / ^ \

**Keys that are used together/similar, live together.**
- ```.``` and ```?``` share the same key (?.)
- ```*``` and ```%``` share the same key (they are both wildcards)
- ```!``` and ```=``` share the same key.
- ```<=>``` are right together so it makes ```==``` ```<=``` ```>=``` and ```=>``` easy.
- $@ are next to each other
- ```|``` and ```&``` are next to each other
- ``````` and ```#``` are on the same key

**The priorities of some keys is changed.**
- On the quote key, double quote is default and you shift for single quote.
- On the underscore key, underscore is default and you shift for minus

**Special ligatures**
If you notice, some keys have subtext that allows really quick typing.  For example, if I press ```CONTROL+ALT+C, I, H``` then it will output ```internal abstract class```

The only thing I'm not completely sure about is that I feel like maybe I should swap ```[ ! ]``` with ```< = >``` (invert their shift) but I'm not sure.

Anyway, what do you think?  I'd love any thoughts or suggestions.

**Comment by TonyValenti** [comment]
I am happy to announce the final version of Dvorak#.

![image](https://user-images.githubusercontent.com/13609558/232602485-04ddf077-e6b7-4bf7-8132-92a7b90d379a.jpeg)

**Reply by HaloFour** [reply]
Going for a Severance look?

**Reply by TonyValenti** [reply]
I dont know what Severance is.  Can you post a link?
I just picked colors I thought looked good together.

**Reply by orthoxerox** [reply]
What are the reasons for the changes between v1 and v2?

**Reply by HaloFour** [reply]
It's a creepy drama on AppleTV about employees who have their memories split so that they can't recall anything that happened in the office once they leave for the day and vice versa.  They had very interesting looking terminals, kinda retro-ish:

https://blog.adafruit.com/2022/03/28/severance-prop-terminals-appletv-severance/

![image](https://user-images.githubusercontent.com/10102290/232618641-e191f65d-90bc-43c8-89b4-587e2d1b1c76.png)

**Reply by CyrusNajmabadi** [reply]
Creepy?  It seems idyllic to me! :-)

**Reply by HaloFour** [reply]
Have you watched the series? :)

**Reply by TonyValenti** [reply]
@orthoxerox V2 was really about two things:
1. Reposition a few keys based on frequency
2. Optimizing placement of common "blends" so that they can be done in one hand stroke.

As a few examples:
1. ```$@"``` can now be hit really quickly
2. Combos like ```=>``` and ```=<``` no longer require a ```shift```
3. Combos like ```:\``` now exist
5. I really didnt like the visuals on the top row numbers alternating.
6. ```&``` and ```|``` were deprioritized based on usage

**Reply by CyrusNajmabadi** [reply]
> Have you watched the series? :)

Yup.  Loved it.  Nice peaceful office (minus the goats) where you could be usper productive.

**Reply by HaloFour** [reply]
@CyrusNajmabadi 

Have you earned that waffle party?
